main.py

```diff
@@ -1,10 +1,70 @@
 import init_django_orm  # noqa: F401
 
+import json
+
 from db.models import Race, Skill, Player, Guild
 
 
+def add_race(player: dict) -> None:
+    player_race = player.get("race")
+    race_name = player_race.get("name")
+    race_description = player_race.get("description")
+
+    if not Race.objects.filter(name=race_name).exists():
+        Race.objects.create(name=race_name, description=race_description)
+
+
+def add_skill(player: dict) -> None:
+    skills = player["race"]["skills"]
+    race_name = player["race"]["name"]
+
+    for skill in skills:
+        if not Skill.objects.filter(name=skill["name"]).exists():
+            Skill.objects.create(
+                name=skill["name"],
+                bonus=skill["bonus"],
+                race=Race.objects.get(name=race_name)
+            )
+
+
+def add_guild(player: dict) -> None:
+    guild = player.get("guild")
+
+    if guild:
+        guild_name = guild.get("name")
+        guild_description = guild.get("description")
+
+        if not Guild.objects.filter(name=guild_name).exists():
+            Guild.objects.create(
+                name=guild_name,
+                description=guild_description
+            )
```

Comment:
You have a lot of .exists() search which have complicity O(n) compare to  get_or_create with the time complexity is typically O(log n) or O(1) because of the indexing. Also there is three independent loop with O(n) each. With big data all this code will run to slow

Reply:
it makes sense thanks